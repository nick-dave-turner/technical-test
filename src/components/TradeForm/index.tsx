import React from "react";
import styled from "styled-components";

import Widget from "../Widget";

const FormRow = styled.div`
  margin-top: 10px;
  &:first-child {
    margin: 0;
  }
`;

const Input = styled.input`
  width: 100%;
  padding: 0.625rem;
  border-radius: 0.625rem;
  border: 0 none;
  background: #ededed;
`;

const Select = styled.select`
  width: 100%;
  height: 38px;
  padding: 0.625rem;
  border-radius: 0.625rem;
  border: 0 none;
  background: #ededed;
`;

const Button = styled.button`
  width: 100%;
  padding: 0.625rem;
  border-radius: 0.625rem;
  border: 0 none;
  background: #337ab7;
  color: #ffffff;
  cursor: pointer;
  &[disabled] {
    opacity: 0.5;
  }
`;

type TradeFormProps = {
  symbols: Array<IStockSymbol>;
};

const TradeForm: React.FC<TradeFormProps> = ({ symbols }) => {
  return (
    <Widget title="Submit Trade">
      <form>
        <FormRow>
          <Select placeholder="Stock" name="stockSymbol">
            <option>Choose Stock</option>
            {symbols.map((symbol: IStockSymbol) => {
              return (
                <option key={symbol.id} value={symbol.stockSymbol}>
                  {symbol.stockSymbol}
                </option>
              );
            })}
          </Select>
        </FormRow>
        <FormRow>
          <Input placeholder="Price" name="price" type="number" />
        </FormRow>
        <FormRow>
          <Input placeholder="Quantity" name="quantity" type="number" />
        </FormRow>
        <FormRow>
          <Button type="button" name="submit-button">
            Submit trade
          </Button>
        </FormRow>
      </form>
    </Widget>
  );
};

export default TradeForm;
