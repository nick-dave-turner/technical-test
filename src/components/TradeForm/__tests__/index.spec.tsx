import React from "react";
import { shallow } from "enzyme";

import TradeForm from "../";

const MOCK_SYMBOLS: Array<IStockSymbol> = [
  {
    id: 1,
    stockSymbol: "TEA"
  },
  {
    id: 2,
    stockSymbol: "POP"
  }
];

describe("TradeForm component", () => {
  describe("When required props are empty", () => {
    it("should match snapshot", () => {
      const component = shallow(<TradeForm symbols={[]} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe("When required props are complete", () => {
    it("should match snapshot", () => {
      const component = shallow(<TradeForm symbols={MOCK_SYMBOLS} />);
      expect(component).toMatchSnapshot();
    });
  });
});
