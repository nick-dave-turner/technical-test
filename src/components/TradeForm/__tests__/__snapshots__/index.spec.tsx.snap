// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TradeForm component When required props are complete should match snapshot 1`] = `
<Widget
  title="Submit Trade"
>
  <form>
    <styled.div>
      <styled.select
        name="stockSymbol"
        placeholder="Stock"
      >
        <option>
          Choose Stock
        </option>
        <option
          key="1"
          value="TEA"
        >
          TEA
        </option>
        <option
          key="2"
          value="POP"
        >
          POP
        </option>
      </styled.select>
    </styled.div>
    <styled.div>
      <styled.input
        name="price"
        placeholder="Price"
        type="number"
      />
    </styled.div>
    <styled.div>
      <styled.input
        name="quantity"
        placeholder="Quantity"
        type="number"
      />
    </styled.div>
    <styled.div>
      <styled.button
        name="submit-button"
        type="button"
      >
        Submit trade
      </styled.button>
    </styled.div>
  </form>
</Widget>
`;

exports[`TradeForm component When required props are empty should match snapshot 1`] = `
<Widget
  title="Submit Trade"
>
  <form>
    <styled.div>
      <styled.select
        name="stockSymbol"
        placeholder="Stock"
      >
        <option>
          Choose Stock
        </option>
      </styled.select>
    </styled.div>
    <styled.div>
      <styled.input
        name="price"
        placeholder="Price"
        type="number"
      />
    </styled.div>
    <styled.div>
      <styled.input
        name="quantity"
        placeholder="Quantity"
        type="number"
      />
    </styled.div>
    <styled.div>
      <styled.button
        name="submit-button"
        type="button"
      >
        Submit trade
      </styled.button>
    </styled.div>
  </form>
</Widget>
`;
